templates/user/view/block/user-name.php

```diff
@@ -3,7 +3,12 @@
 defined( 'ABSPATH' ) || exit;
 
 $display = get_option( 'hp_user_enable_display' );
+$online_users = (array) get_transient( 'online_users' );
+?>
 
+<span class="hp-vendor__online-status <?php echo in_array( $user->get_id(), array_keys( $online_users ) ) && $online_users[ $user->get_id() ] > time() - 60 ? 'online' : 'offline'; ?>"></span>
```

Comment:
Checking the time-60 is logic, templates should ideally only contain presentation layer (markup, styles, inserted variables, conditions). Please move this logic to the code which defines which users are considered online and those considered offline shouldn't be in the array at all.

After moving set_transient occurances to the request context callback you can add a code there that will go through the IDs and check the time, and update the transient if some IDs should be removed (those older than 5 minutes). So on every front-end website visit the array will be cleaned up if required, and in the template code you can just check if the user ID is among the stored IDs.